From 27b1b767d162ecc55afc899c7f523669de95c190 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Wed, 4 Dec 2024 09:35:00 +0100
Subject: [PATCH] phy: stm32: fix aarch32 build for STM32 USB2-FEMTO PHY
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix aarch32 build error:

error: implicit declaration of function ‘FIELD_FIT’
 [-Werror=implicit-function-declaration]
         if (!ret && FIELD_FIT(SYSCFG_USB2PHYTRIM1_PLLITUNE_MASK, val)) {
  |                             ^~~~~~~~~
error: implicit declaration of function ‘FIELD_PREP’
 [-Werror=implicit-function-declaration]
phy_dev->value_trim1 |= FIELD_PREP(SYSCFG_USB2PHYTRIM1_PLLITUNE_MASK, val);

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: I147a7b069413e7dc7a31d07db0b9421504ca455d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/422523
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/phy/st/phy-stm32-usb2phy.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/phy/st/phy-stm32-usb2phy.c
+++ b/drivers/phy/st/phy-stm32-usb2phy.c
@@ -6,6 +6,7 @@
  * Copyright (C) 2022 STMicroelectronics
  * Author(s): Pankaj Dev <pankaj.dev@st.com>.
  */
+#include <linux/bitfield.h>
 #include <linux/clk.h>
 #include <linux/clk-provider.h>
 #include <linux/io.h>
